src/connections/SitePolygons.ts

```diff
@@ -1,76 +1,84 @@
-import { useMemo } from "react";
 import { createSelector } from "reselect";
 
-import {
-  sitePolygonsIndex,
-  SitePolygonsIndexQueryParams
-} from "@/generated/v3/researchService/researchServiceComponents";
-import {
-  sitePolygonsIndexFetchFailed,
-  sitePolygonsIndexIsFetching
-} from "@/generated/v3/researchService/researchServicePredicates";
+import { sitePolygonsIndex } from "@/generated/v3/researchService/researchServiceComponents";
+import { sitePolygonsIndexFetchFailed } from "@/generated/v3/researchService/researchServicePredicates";
 import { SitePolygonDto } from "@/generated/v3/researchService/researchServiceSchemas";
-import { ApiDataStore } from "@/store/apiSlice";
+import ApiSlice, { ApiDataStore, indexMetaSelector, PendingErrorState } from "@/store/apiSlice";
 import { Connection } from "@/types/connection";
 import { connectionHook, connectionLoader } from "@/utils/connectionShortcuts";
+import { selectorCache } from "@/utils/selectorCache";
 
 export type SitePolygonConnectionParams = {
   entityName: string;
   entityUuid: string;
 };
 
-export type SitePolygonConnection = {
-  sitePolygons?: SitePolygonDto[] | undefined;
-  sitePolygonsHaveFailed: boolean;
-  sitePolygonsAreLoading: boolean;
+export type SitePolygonIndexConnectionProps = SitePolygonConnectionParams & {
+  pageSize?: number;
+  pageNumber?: number;
 };
 
-const sitePolygonQuery = (entityName: string, entityUuid: string) => {
-  const queryParams = {
-    "page[number]": 0, // update once pagination is implemented
-    "page[size]": 10
-  } as SitePolygonsIndexQueryParams;
-  if (entityName === "projects") {
-    queryParams["projectId[]"] = [entityUuid];
-  } else if (entityName === "sites") {
-    queryParams["siteId[]"] = [entityUuid];
-  }
-  return { queryParams };
+const ENTITY_QUERY_KEYS: Record<string, string> = {
+  projects: "projectId[]",
+  sites: "siteId[]"
 };
-const sitePolygonsSelector = (store: ApiDataStore) =>
-  Object.values(store.sitePolygons || {}).map(resource => resource.attributes);
 
-const sitePolygonsConnection = (params: SitePolygonConnectionParams): Connection<SitePolygonConnection> => ({
-  load: ({ sitePolygons }) => {
-    if (sitePolygons == null || sitePolygons.length === 0) {
-      const queryParams = sitePolygonQuery(params.entityName, params.entityUuid);
-      sitePolygonsIndex(queryParams);
+export type SitePolygonIndexConnection<SitePolygonDto> = {
+  sitePolygons?: SitePolygonDto[];
+  indexTotal?: number;
+  fetchFailure?: PendingErrorState | null;
+  refetch: () => void;
```

Comment:
are indexTotal and refetch useful to the consumers of this connection? There doesn't seem to be any code using them in this PR.